daphne/src/roles_test.rs

```diff
@@ -20,6 +23,18 @@ use prio::codec::{Decode, Encode};
 use rand::{thread_rng, Rng};
 use std::{ops::DerefMut, vec};
 
+fn gen_test_upload_req(agg: &MockAggregator, report: Report) -> DapRequest<BearerToken> {
```

Comment:
Suggestion (here and below): It would be a bit cleaner to define this function as part of MockAggregator's implementation:

```rust
impl MockAggregator {
    fn gen_test_upload(&self, report: Report) -> DapRequest<BearerToken> {
        ...
    }
}
```

Reply:
Great suggestion, I think this would make the code a bit more readable as well.